### PR TITLE
[SYCL-MLIR] Drop `-sycl-method-to-sycl-call` pass

### PR DESCRIPTION
**Pull request by victor-eds**
This pass is no longer needed as all "SYCLMethodOps" are now supported by `-convert-sycl-to-llvm`.

### CODE REVIEW

File: polygeist/tools/cgeist/driver.cc [hunk 1]
```diff
@@ -667,8 +667,6 @@ static LogicalResult finalize(mlir::MLIRContext &Ctx,
       // PM3.addPass(mlir::createLowerFuncToLLVMPass(options));
       PM3.addPass(polygeist::createLegalizeForSPIRVPass());
 
-      // Needed because SYCLMethodOps lowering might introduce redundant
-      // operations.
       PM3.addPass(mlir::createCSEPass());
```
Comment (whitneywhtsang) [inline polygeist/tools/cgeist/driver.cc:670]: CSE pass can be removed as well?

Comment (victor-eds) [reply polygeist/tools/cgeist/driver.cc:670]: I'd keep it, as our lowering might be introducing redundant operations anyway

**Review by whitneywhtsang — APPROVED** [review]

File: polygeist/tools/cgeist/driver.cc (hunk 1 above)
Comment (whitneywhtsang) [reply polygeist/tools/cgeist/driver.cc:670]: ok

**Review by etiotto — APPROVED** [review]
Nice cleanup!

**Pull request merged**